engine/src/main/java/org/hibernate/search/engine/search/aggregation/spi/AggregationTypeKeys.java

```diff
@@ -18,5 +18,17 @@ private AggregationTypeKeys() {
 			of( IndexFieldTraits.Aggregations.TERMS );
 	public static final SearchQueryElementTypeKey<RangeAggregationBuilder.TypeSelector> RANGE =
 			of( IndexFieldTraits.Aggregations.RANGE );
+	public static final SearchQueryElementTypeKey<MetricFieldAggregationBuilder.TypeSelector> SUM =
+			of( IndexFieldTraits.Aggregations.SUM );
+	public static final SearchQueryElementTypeKey<MetricFieldAggregationBuilder.TypeSelector> MIN =
+			of( IndexFieldTraits.Aggregations.MIN );
+	public static final SearchQueryElementTypeKey<MetricFieldAggregationBuilder.TypeSelector> MAX =
+			of( IndexFieldTraits.Aggregations.MAX );
+	public static final SearchQueryElementTypeKey<SearchAggregationFilteredBuilder<Long>> COUNT =
+			of( IndexFieldTraits.Aggregations.COUNT );
+	public static final SearchQueryElementTypeKey<SearchAggregationFilteredBuilder<Long>> COUNT_DISTINCT =
+			of( IndexFieldTraits.Aggregations.COUNT_DISTINCT );
+	public static final SearchQueryElementTypeKey<SearchAggregationFilteredBuilder<Double>> AVG =
```

Comment:
This seems wrong: as a user, I'd expect an average on a `LocalTime` field to result in a `LocalTime` value. A `Double` would not mean anything.

IMO this case is very similar to `sum`, where you decided to go for `MetricFieldAggregationBuilder` (which I agree with), and you should do the same here.

Reply:
I think you're right. Going to changing it to use `MetricFieldAggregationBuilder.TypeSelector` as sum.

Reply:
I think you're right. Going to changing it to use `MetricFieldAggregationBuilder.TypeSelector` as sum.

Reply:
I think you're right. Going to changing it to use `MetricFieldAggregationBuilder.TypeSelector` as sum.

Reply:
I think you're right. Going to changing it to use `MetricFieldAggregationBuilder.TypeSelector` as sum.

Reply:
🙈🙈🙈🙈😃

Reply:
@yrodiere changing the `avg` implementation from `ElasticsearchMetricDoubleAggregation` to `ElasticsearchMetricFieldAggregation`, it means that if for instance the field has type `Integer` the average will become an integer, losing the floating point values. I'm not sure it is actually a good idea. Anyway I'll keep this change in a different commit, so that we could easily rollback it in case.

Reply:
Makes sense. Then maybe let's not provide a default? Change the DSL to force users to provide a type?